components/brave_rewards/resources/page/components/adsBox.tsx

```diff
@@ -175,71 +175,95 @@ class AdsBox extends React.Component<Props, State> {
-      return {
-        id: item.id,
-        date: new Date(item.timestampInMilliseconds).toLocaleDateString(),
-        adDetailRows: (
-          item.adDetailRows.map((itemDetail: Rewards.AdHistoryDetail) => {
-            let brandInfo = itemDetail.adContent.brandInfo
-            if (brandInfo.length > 50) {
-              brandInfo = brandInfo.substring(0, 50) + '...'
-            }
-            const adContent: Rewards.AdContent = {
-              uuid: itemDetail.adContent.uuid,
-              creativeSetId: itemDetail.adContent.creativeSetId,
-              brand: itemDetail.adContent.brand,
-              brandInfo: brandInfo,
-              brandLogo: itemDetail.adContent.brandLogo,
-              brandDisplayUrl: itemDetail.adContent.brandDisplayUrl,
-              brandUrl: itemDetail.adContent.brandUrl,
-              likeAction: itemDetail.adContent.likeAction,
-              adAction: itemDetail.adContent.adAction,
-              savedAd: itemDetail.adContent.savedAd,
-              flaggedAd: itemDetail.adContent.flaggedAd,
-              onThumbUpPress: () =>
-                this.onThumbUpPress(itemDetail.adContent.uuid,
-                                    itemDetail.adContent.creativeSetId,
-                                    itemDetail.adContent.likeAction),
-              onThumbDownPress: () =>
-                this.onThumbDownPress(itemDetail.adContent.uuid,
-                                      itemDetail.adContent.creativeSetId,
-                                      itemDetail.adContent.likeAction),
-              onMenuSave: () =>
-                this.onMenuSave(itemDetail.adContent.uuid,
-                                itemDetail.adContent.creativeSetId,
-                                itemDetail.adContent.savedAd),
-              onMenuFlag: () =>
-                this.onMenuFlag(itemDetail.adContent.uuid,
-                                itemDetail.adContent.creativeSetId,
-                                itemDetail.adContent.flaggedAd)
-            }
-            const categoryContent: Rewards.CategoryContent = {
-              category: itemDetail.categoryContent.category,
-              optAction: itemDetail.categoryContent.optAction,
-              onOptInAction: () =>
-                this.onOptInAction(itemDetail.categoryContent.category,
-                                   itemDetail.categoryContent.optAction),
-              onOptOutAction: () =>
-                this.onOptOutAction(itemDetail.categoryContent.category,
-                                    itemDetail.categoryContent.optAction)
-            }
-            return {
-              id: itemDetail.id,
-              adContent: adContent,
-              categoryContent: categoryContent
-            }
+  getGroupedAdsHistory = (adsHistory: Rewards.AdsHistory[], savedOnly: boolean) => {
+    let groupedAdsHistory: Rewards.AdsHistory[] = []
+
+    for (let i = 0; i < adsHistory.length; i++) {
+      const adHistory = adsHistory[i]
```

Comment:
could this be null/undefined? I think it would be good to check it

Reply:
This cannot be null or undefined